hw_strings.py

```diff
@@ -0,0 +1,178 @@
+import re
+import pprint
+from collections import Counter
+from csv import DictWriter
+
+
+# 1)Из текстового файла удалить все слова, содержащие от трех до пяти символов,
+# но при этом из каждой строки должно быть удалено только четное количество таких слов.
+
+def remove_even_words_of_length(src, target, minval, maxval=None):
+    """Function takes .txt file (src) and removes even occurences of words with required length per line.
+     minval - specifies min length of a word.
+     maxval defaults to minval. If specified, length is considered to be [minval, maxval] including border values.
+     target - target file to save selected lines to"""
+    min_len = minval
+    if maxval is None:
+        max_len = min_len
+    else:
+        max_len = maxval
+    with open(src, encoding='utf-8') as src_file, open(target, "w+", encoding='utf-8') as target_file:
```

Comment:
This comment can be applied to all the functions. Do not forget to check if the file exists before opening it.